datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -2799,14 +2817,71 @@ impl PartialOrd for Extension {
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Hash)]
 pub struct Limit {
     /// Number of rows to skip before fetch
-    pub skip: usize,
+    pub skip: Option<Box<Expr>>,
```

Comment:
Would it be possible to do the constant folding when _building_ Limit node, so that logical plan structure remains intact? See also https://github.com/apache/datafusion/issues/12723

Reply:
> What are the constraints on the expression that can be used here? For example, can it have any column references?

I think it can be any integer expression, and can also contain column references. Both PostgreSQL and DuckDB support `select 1 limit (select max(col0) from t)`.
```sh
v1.1.1-dev319 af39bd0dcf
D create table t as values(1);
D select 1 limit (select max(col0) from t);
┌───────┐
│   1   │
│ int32 │
├───────┤
│     1 │
└───────┘
```